### ISSUE TITLE
Add test for async functions

### ISSUE THREAD

**Issue by cjihrig** [hapijs/code#104]
This test works, but currently fails linting (the parser fails). Also have to deal with Node version issues.

```js
it('validates correct type for async functions', (done) => {

    let exception = false;
    try {
        Code.expect(async function () {}).to.be.a.function();
    }
    catch (err) {
        exception = err;
    }

    Hoek.assert(!exception, exception);
    done();
});
```

**Comment by cjihrig** [comment]
Adding this to the 5.0.0 milestone because lab has been updated.

**Comment by cjihrig** [comment]
Closed in bb37cc55a8a03c27df1568ad7377ff6565cd9149.
